aws/resource_aws_codedeploy_app.go

```diff
@@ -101,52 +110,74 @@ func resourceAwsCodeDeployAppCreate(d *schema.ResourceData, meta interface{}) er
 	// the state file. This allows us to reliably detect both when the TF
 	// config file changes and when the user deletes the app without removing
 	// it first from the TF config.
-	d.SetId(fmt.Sprintf("%s:%s", *resp.ApplicationId, application))
+	d.SetId(fmt.Sprintf("%s:%s", aws.StringValue(resp.ApplicationId), application))
 
 	return resourceAwsCodeDeployAppRead(d, meta)
 }
 
 func resourceAwsCodeDeployAppRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).codedeployconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
 
 	application := resourceAwsCodeDeployAppParseId(d.Id())
```

Comment:
to support a change in ID suggested in the comment below:
```suggestion
        if !d.IsNewResource() && !strings.Contains(d.Id(), d.Get("name").(string)) {
		d.SetId(fmt.Sprintf("%s:%s", d.Get("application_id").(string), d.Get("name").(string)))
	}
	application := resourceAwsCodeDeployAppParseId(d.Id())
```